lib/split/experiment.rb

```diff
@@ -35,8 +36,8 @@ def initialize(name, options = {})
       set_alternatives_and_options(options)
     end
 
-    def self.finished_key(key)
-      "#{key}:finished"
+    def self.finished_key(key, version_key = version)
+      "#{key.split(":")[0..-2].join}:#{version_key}:finished"
```

Comment:
1) This could be more robust against bad user input. For instance, assuming that there will be a `:` char present in the key.
2) Also worth adding tests for this new behavior + the handling of bad input.